### PR TITLE
test(artifacts): Added an ubuntu 20 with FIPS enabled scenario

### PR DESCRIPTION
**Pull request by ShlomiBalalis**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: test-cases/artifacts/ubuntu2004-fips.yaml
Comment (fruch) [inline test-cases/artifacts/ubuntu2004-fips.yaml:18]: And this work? I'm quite surprised it would without some work on c-s calls

Comment (roydahan) [reply test-cases/artifacts/ubuntu2004-fips.yaml:18]: Actually we don't have clients in this case, so I'm not sure how valuable it is.

Comment (ShlomiBalalis) [reply test-cases/artifacts/ubuntu2004-fips.yaml:18]: > And this work? I'm quite surprised it would without some work on c-s calls

No no, hence (partially) the Draft. The c-s fails

> Actually we don't have clients in this case, so I'm not sure how valuable it is.

We actually do have 2 stresses that are executed 3 times: https://github.com/scylladb/scylla-cluster-tests/blob/master/artifacts_test.py#L123

Comment (fruch) [reply test-cases/artifacts/ubuntu2004-fips.yaml:18]: And cqlsh calls, and few calls connecting with python-driver.

So yeah we do have clients

**Comment by ShlomiBalalis** [comment]
https://jenkins.scylladb.com/job/scylla-staging/job/Shlomo/job/artifacts-ubuntu2004-FIPS-test/21/ Passing run

File: test-cases/artifacts/ubuntu2004-fips.yaml
Comment (fruch) [inline test-cases/artifacts/ubuntu2004-fips.yaml:15]: I think those defaults should go, if someone wants to run it he would need to supply the repo

**Comment by fgelcer** [comment]
> https://jenkins.scylladb.com/job/scylla-staging/job/Shlomo/job/artifacts-ubuntu2004-FIPS-test/21/ Passing run

i wonder, how can we be sure that FIPS is enabled?

File: test-cases/artifacts/ubuntu2004-fips.yaml
Comment (ShlomiBalalis) [reply test-cases/artifacts/ubuntu2004-fips.yaml:15]: By "those" you mean the repo only, right? Maybe the region, too, but not the rest

**Comment by ShlomiBalalis** [comment]
> > https://jenkins.scylladb.com/job/scylla-staging/job/Shlomo/job/artifacts-ubuntu2004-FIPS-test/21/ Passing run
> 
> i wonder, how can we be sure that FIPS is enabled?

It's basically part of the image, but I can add a check. Where would it be? Inside the test function?

**Comment by fgelcer** [comment]
> > > https://jenkins.scylladb.com/job/scylla-staging/job/Shlomo/job/artifacts-ubuntu2004-FIPS-test/21/ Passing run
> > 
> > 
> > i wonder, how can we be sure that FIPS is enabled?
> 
> It's basically part of the image, but I can add a check. Where would it be? Inside the test function?

first, how can one confirm it is enabled.. i'm not sure we need to add it to the test (as it should be part of the image), but perhaps we need to confirm FIPS is working, before we run a test and consider it passed... WDYT?

**Comment by ShlomiBalalis** [comment]
> > > > https://jenkins.scylladb.com/job/scylla-staging/job/Shlomo/job/artifacts-ubuntu2004-FIPS-test/21/ Passing run
> > > 
> > > 
> > > i wonder, how can we be sure that FIPS is enabled?
> > 
> > 
> > It's basically part of the image, but I can add a check. Where would it be? Inside the test function?
> 
> first, how can one confirm it is enabled.. i'm not sure we need to add it to the test (as it should be part of the image), but perhaps we need to confirm FIPS is working, before we run a test and consider it passed... WDYT?

There's a command for it:
```
ubuntu@ip-10-0-3-78:~$ pro status --all
SERVICE          ENTITLED  STATUS    DESCRIPTION
esm-apps         yes       enabled   Expanded Security Maintenance for Applications
esm-infra        yes       enabled   Expanded Security Maintenance for Infrastructure
fips             yes       enabled   NIST-certified core packages
fips-updates     yes       disabled  NIST-certified core packages with priority security updates
livepatch        yes       n/a       Canonical Livepatch service
usg              yes       disabled  Security compliance and audit tools
```
Maybe in the node_setup?

**Comment by ShlomiBalalis** [comment]
Also, please don't merge yet. I want to add more than encryption types

**Comment by fruch** [comment]
> And this command as well:
> ```
> ubuntu@ip-10-0-3-78:~$ cat /proc/sys/crypto/fips_enabled
> 1
> ```


This seems to be the right way to confirm:
https://access.redhat.com/discussions/3293631#comment-1251771

**Comment by ShlomiBalalis** [comment]
> > And this command as well:
> > ```
> > ubuntu@ip-10-0-3-78:~$ cat /proc/sys/crypto/fips_enabled
> > 1
> > ```
> 
> This seems to be the right way to confirm: https://access.redhat.com/discussions/3293631#comment-1251771

Why go to redhead when you can go to the source?
https://ubuntu.com/tutorials/using-the-ua-client-to-enable-fips#4-enabling-fips-crypto-modules

**Comment by fruch** [comment]
> > > And this command as well:
> > > ```
> > > ubuntu@ip-10-0-3-78:~$ cat /proc/sys/crypto/fips_enabled
> > > 1
> > > ```
> > 
> > This seems to be the right way to confirm: https://access.redhat.com/discussions/3293631#comment-1251771
> 
> Why go to redhat when you can go to the source?
> https://ubuntu.com/tutorials/using-the-ua-client-to-enable-fips#4-enabling-fips-crypto-modules

I go where google take me, and if that where are redheads, there I'll go :)

File: test-cases/artifacts/ubuntu2004-fips.yaml
Comment (ShlomiBalalis) [reply test-cases/artifacts/ubuntu2004-fips.yaml:18]: > And cqlsh calls, and few calls connecting with python-driver.
> 
> So yeah we do have clients

I updated the stresses and now they work, but I didn't change anything regarding any "cqlsh calls". Is it possible that it happens automatically?

**Comment by ShlomiBalalis** [comment]
> Also, please don't merge yet. I want to add more than encryption types

Added server_encrypt, internode_encryption and scylla_encryption_options (encryption at rest, copied from `local-ear.yaml`)

File: test-cases/artifacts/ubuntu2004-fips.yaml
Comment (ShlomiBalalis) [reply test-cases/artifacts/ubuntu2004-fips.yaml]: Removed the scylla_repo

**Comment by ShlomiBalalis** [comment]
> > Also, please don't merge yet. I want to add more than encryption types
> 
> Added server_encrypt, internode_encryption and scylla_encryption_options (encryption at rest, copied from `local-ear.yaml`)

I saw that the `local-ear.yaml` file is a configuration file, so I added it the jenkinsfile instead of its contents

File: artifacts_test.py
Comment (roydahan) [inline artifacts_test.py:117]: weird that we have another implementation also here.
We probably should unify them, but not critical for this PR.

**Review by roydahan — DISMISSED** [review]

File: artifacts_test.py
Comment (fruch) [reply artifacts_test.py:117]: Since artifacts tests are not using loader nodes, aligning it won't be that straight forward, and not necessarily worth the effort.

For example calling the CassandraStress SCT class would run c-s in docker by default.

**Comment by fruch** [comment]
> > > Also, please don't merge yet. I want to add more than encryption types
> > 
> > Added server_encrypt, internode_encryption and scylla_encryption_options (encryption at rest, copied from `local-ear.yaml`)
> 
> I saw that the `local-ear.yaml` file is a configuration file, so I added it the jenkinsfile instead of its contents

This test case is gonna run only on  enterprise versions ? Since EaR is enterprise only feature...

**Comment by ShlomiBalalis** [comment]
> > > > Also, please don't merge yet. I want to add more than encryption types
> > > 
> > > 
> > > Added server_encrypt, internode_encryption and scylla_encryption_options (encryption at rest, copied from `local-ear.yaml`)
> > 
> > 
> > I saw that the `local-ear.yaml` file is a configuration file, so I added it the jenkinsfile instead of its contents
> 
> This test case is gonna run only on enterprise versions ? Since EaR is enterprise only feature...

Fips support is also an enterprise feature, so it lines up!

**Comment by ShlomiBalalis** [comment]
> > > > > https://jenkins.scylladb.com/job/scylla-staging/job/Shlomo/job/artifacts-ubuntu2004-FIPS-test/21/ Passing run
> > > > 
> > > > 
> > > > i wonder, how can we be sure that FIPS is enabled?
> > > 
> > > 
> > > It's basically part of the image, but I can add a check. Where would it be? Inside the test function?
> > 
> > 
> > first, how can one confirm it is enabled.. i'm not sure we need to add it to the test (as it should be part of the image), but perhaps we need to confirm FIPS is working, before we run a test and consider it passed... WDYT?
> 
> There's a command for it:
> 
> ```
> ubuntu@ip-10-0-3-78:~$ pro status --all
> SERVICE          ENTITLED  STATUS    DESCRIPTION
> esm-apps         yes       enabled   Expanded Security Maintenance for Applications
> esm-infra        yes       enabled   Expanded Security Maintenance for Infrastructure
> fips             yes       enabled   NIST-certified core packages
> fips-updates     yes       disabled  NIST-certified core packages with priority security updates
> livepatch        yes       n/a       Canonical Livepatch service
> usg              yes       disabled  Security compliance and audit tools
> ```
> 
> Maybe in the node_setup?

@fgelcer please confirm, should I verify on node_setup? Can't think about another place other than the test

**Comment by fgelcer** [comment]
> @fgelcer please confirm, should I verify on node_setup? Can't think about another place other than the test

during `node_setup` sounds to be the right place... but i'm not sure you can raise an event from there... if not, the error message must be very clear

**Comment by ShlomiBalalis** [comment]
> > @fgelcer please confirm, should I verify on node_setup? Can't think about another place other than the test
> 
> during `node_setup` sounds to be the right place... but i'm not sure you can raise an event from there... if not, the error message must be very clear

@fgelcer Added

File: sdcm/cluster.py
Comment (fruch) [inline sdcm/cluster.py:4373]: are we should this command is available on other ubuntu ? did we tested it on any other ubuntu artifact test ?

**Review by fruch — APPROVED** [review]
LGTM

File: sdcm/cluster.py
Comment (ShlomiBalalis) [reply sdcm/cluster.py:4373]: It should be according to what I understand from the website, but I'll run the ubuntu 20 and 22 artifacts to be sure

**Comment by ShlomiBalalis** [comment]
Passing job: https://jenkins.scylladb.com/job/scylla-staging/job/Shlomo/job/artifacts-ubuntu2004-FIPS-test/25/

File: sdcm/cluster.py
Comment (ShlomiBalalis) [reply sdcm/cluster.py:4373]: [Done](https://jenkins.scylladb.com/job/scylla-staging/job/Shlomo/job/artifacts-ubuntu2004-test-FIPS-branch/2)
and
[Done](https://jenkins.scylladb.com/job/scylla-staging/job/Shlomo/job/artifacts-ubuntu2204-test-FPIS-brnach/2)

Comment (fruch) [reply sdcm/cluster.py:4373]: No

**Review by fgelcer — APPROVED** [review]
added a filter to FIPS in `create-test-release-jobs`, as this test should not be created to any OSS.

besides that, LGTM

**Comment by ShlomiBalalis** [comment]
@fruch Can you re-approve?

File: sct.py
Comment (fruch) [inline sct.py:1382]: if it's filtered here, where exactly it's gonna be generated ?
i.e. there no enterprise only artifacts jobs till now

**Review by fruch — APPROVED** [review]
LGTM

**Comment by fgelcer** [comment]
@fruch , it wasn't set, but they want to backport the FIPS feature to 2023.1... so we need this to be backported as well

File: sct.py
Comment (fgelcer) [reply sct.py:1382]: well, if it will not be created, we will need to add in the enterprise code to add also artifacts (like we have for upgrade, and longevity jobs)

**Comment by ShlomiBalalis** [comment]
@scylladb/qa-maintainers Can anyone please backport?

**Comment by roydahan** [comment]
Backporting

**Comment by roydahan** [comment]
Backported to 2023.1

**Pull request merged**
